### PR TITLE
Creation of basic user and appointment interfaces

### LINKED ISSUE #8: Create basic interfaces for User and Appointment, so everyone can start work on their task.

### PR DESCRIPTION
**Pull request by MrZaiko**
These interfaces will be the basis for all the upcoming work

Resolves: #8

### CODE REVIEW

**Comment by tombenamato** [comment]
For the Appointement class, I think we should also have a function to remove a student from it

**Comment by tombenamato** [comment]
I just want to point out that for the user class as it is now, we can not change the name or the surname. However since it is just not present this should'nt be a problem. As when we will want to add this functionallity we could just add it.

**Comment by tombenamato** [comment]
> For the Appointement class, I think we should also have a function to remove a student from it

As you can see with the other comment, it is not that big a problem we could just add it as we go.

**Comment by tombenamato** [comment]
For the function returning time in the Appointment class, in the documentation so that it is clear for everyone you should precise that the time is in Millisecond like it is represented for android in general.

**Review by tombenamato — COMMENTED** [review]
For the User class, the getId function. see https://stackoverflow.com/questions/932102/what-data-type-is-recommended-for-id-columns. There is not better way but it is still worth noting the pros and cons.

File: app/src/main/java/io/github/polysmee/interfaces/Appointment.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package io.github.polysmee.interfaces;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A generic appointment
+ */
+public interface Appointment {
```
Comment (Morpheo1) [inline app/src/main/java/io/github/polysmee/interfaces/Appointment.java:8]: The appointment interface seems to have all the necessary methods

File: app/src/main/java/io/github/polysmee/interfaces/User.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package io.github.polysmee.interfaces;
+
+import java.util.List;
+
+/**
+ * A generic user
+ */
+public interface User {
```
Comment (Morpheo1) [inline app/src/main/java/io/github/polysmee/interfaces/User.java:8]: Same for the User interface

**Review by Morpheo1 — APPROVED** [review]

**Review by tombenamato — CHANGES_REQUESTED** [review]
The Documentation for time return function in Appointment

File: app/src/main/java/io/github/polysmee/interfaces/Appointment.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Adds the given user to the list of participant
+     * @param newParticipant the user to be added
+     */
+    void addParticipant(User newParticipant);
+
+    /**
+     * Removes the given user to the list of participant
+     * @param participant the user to be removed
+     */
```
Comment (tombenamato) [inline app/src/main/java/io/github/polysmee/interfaces/Appointment.java]: I think a set representation of user would be better than a list. Also see the documentation

File: app/src/main/java/io/github/polysmee/interfaces/User.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     * @param newAppointment the appointment to be added
+     */
+    void addAppointment(Appointment newAppointment);
+
+    /**
+     * Removes the given appointment to the list of appointments
+     * @param appointment the appointment to be removed
+     */
+    void removeAppointment(Appointment appointment);
+}
```
Comment (tombenamato) [inline app/src/main/java/io/github/polysmee/interfaces/User.java]: I think a set representation of user would be better than a list. Also see the documentation

**Review by tombenamato — CHANGES_REQUESTED** [review]

**Review by MRandl — APPROVED** [review]
all good to me for now. We will see if that proves sufficient when the time is needed.

**Review by tombenamato — APPROVED** [review]

**Pull request merged**